src/app/loadout-builder/filter/ExoticPicker.tsx

```diff
@@ -13,48 +16,114 @@ interface Props {
   defs: D2ManifestDefinitions;
   /** A list of item hashes for unlocked exotics. */
   availableExotics?: DimItem[];
+  isPhonePortrait: boolean;
+  language: string;
   lbDispatch: Dispatch<LoadoutBuilderAction>;
   onClose(): void;
 }
 
 /** A drawer to select an exotic for your build. */
-function ExoticPicker({ defs, availableExotics, lbDispatch, onClose }: Props) {
-  const lockableExotics: LockedExotic[] = [];
+function ExoticPicker({
+  defs,
+  availableExotics,
+  isPhonePortrait,
+  language,
+  lbDispatch,
+  onClose,
+}: Props) {
+  const [query, setQuery] = useState('');
 
-  if (availableExotics?.length) {
-    const uniqueExotics = _.uniqBy(availableExotics, (item) => item.hash);
+  const lockableExotics = useMemo(() => {
+    const rtn: LockedExotic[] = [];
 
-    for (const item of uniqueExotics) {
-      const def = defs.InventoryItem.get(item.hash);
+    if (availableExotics?.length) {
+      const uniqueExotics = _.uniqBy(availableExotics, (item) => item.hash);
 
-      if (def?.displayProperties.hasIcon) {
-        lockableExotics.push({ def, bucketHash: item.bucket.hash });
+      for (const item of uniqueExotics) {
+        const def = defs.InventoryItem.get(item.hash);
+
+        if (def?.displayProperties.hasIcon) {
+          rtn.push({ def, bucketHash: item.bucket.hash });
+        }
       }
     }
-  }
+
+    return rtn;
+  }, [availableExotics, defs.InventoryItem]);
+
+  const filteredOrderedAndGroupedExotics = useMemo(() => {
+    // Only some languages effectively use the \b regex word boundary
+    const regexp = ['de', 'en', 'es', 'es-mx', 'fr', 'it', 'pl', 'pt-br'].includes(language)
```

Comment:
we should probably make this a helper instead of duplicating language code 

https://github.com/DestinyItemManager/DIM/blob/cd1dfd969edd7bf7339e41f7c731f4f03dff0755/src/app/search/search-filters/freeform.tsx#L12

Reply:
It's also in the mod picker so I reckon you're right. Want me to do it in this PR or leave it for another?

Reply:
let's leave it for another